app/src/main/java/com/jjswigut/matters/ui/EditMatterFragment.kt

```diff
@@ -34,26 +30,32 @@ class EditMatterFragment : BaseFragment() {
         super.onViewCreated(view, savedInstanceState)
 
         arguments?.let {
-            matter = EditMatterFragmentArgs.fromBundle(it).matter
+            if (it.containsKey(ARG_MATTER)) {
```

Comment:
if the arguments contain the key ARG_MATTER - that is how we will determine if this fragment is being used in the context of creating a new matter or editing an existing one.

you can see we instantiate the `matter` object just below it, so instead of using the arguments from here on out, i am just going to check if that matter object is null. if it is, then i know we are not editing an existing matter